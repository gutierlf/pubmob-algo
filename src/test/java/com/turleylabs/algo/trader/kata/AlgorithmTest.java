package com.turleylabs.algo.trader.kata;

import org.approvaltests.ApprovalUtilities;
import org.approvaltests.Approvals;
import org.junit.Test;

import java.io.ByteArrayOutputStream;
import java.util.Locale;

public class AlgorithmTest {
    @Test
    public void approval_should_pass() {
        Locale.setDefault(Locale.CANADA);
        RefactorMeAlgorithm refactorAlgorithm = new RefactorMeAlgorithm() {
            @Override
            public void initialize() {
                super.initialize();
                super.setEndDate(2012, 03, 06);
            }
        };
        ByteArrayOutputStream out = new ApprovalUtilities().writeSystemOutToStringBuffer();

        refactorAlgorithm.run();

        Approvals.verify(out.toString());
    }
}
