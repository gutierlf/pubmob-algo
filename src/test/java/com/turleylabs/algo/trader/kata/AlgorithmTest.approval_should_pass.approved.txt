2011-01-04 - Buy TQQQ Vix 17.3800. above 10 MA 0.0231
2011-01-14 - Sell TQQQ taking profits. Gain 0.0960. Vix 15.4600
2011-01-25 - Buy TQQQ Vix 17.5900. above 10 MA 0.0007
2011-02-16 - Sell TQQQ taking profits. Gain 0.1259. Vix 16.7200
2011-04-07 - Buy TQQQ Vix 17.1100. above 10 MA 0.0031
2011-05-25 - Sell TQQQ 10 day below 21 day. Gain -0.0300. Vix 17.0700
2011-07-11 - Buy TQQQ Vix 18.3900. above 10 MA 0.0194
2011-07-27 - Sell TQQQ high volatility. Gain 0.0013. Vix 22.9800
2012-01-20 - Buy TQQQ Vix 18.2800. above 10 MA 0.0602
